File: plugins/vehicle_plugins_py/vehicle_plugins_py/motorbike.py
from vehicle_base_py.vehicle_base import VehicleBase

class Motorbike(VehicleBase):
    # --------------------------------------------------------
    # IMPLÉMENTER TOUTES LES MÉTHODES ABSTRAITES
    # --------------------------------------------------------

    def move(self) -> None:
        print("Motorbike is moving fast ! 🏍️")

    def stop(self) -> None:
        print("Motorbike stopped !")

    def get_info(self) -> str:
        return "Motorbike - 2 wheels - fast vehicle"

File: ros2_fundamentals/plugins/vehicle_plugins/src/motorbike.cpp
#include "vehicle_plugins/motorbike.hpp"
#include "pluginlib/class_list_macros.hpp"
//       | macro pour enregistrer le plugin

namespace vehicle_plugins
{

void Motorbike::move()
{
    RCLCPP_INFO(rclcpp::get_logger("Motorbike"), "Motorbike is moving fast ! 🏍️");
}

void Motorbike::stop()
{
    RCLCPP_INFO(rclcpp::get_logger("Motorbike"), "Motorbike stopped !");
}

std::string Motorbike::getInfo()
{
    return "Motorbike - 2 wheels - fast vehicle";
}

}  // namespace vehicle_plugins

// ============================================================
// ENREGISTRER LE PLUGIN
// ============================================================
// PLUGINLIB_EXPORT_CLASS(classe_concrete, classe_de_base)
// Cette macro est OBLIGATOIRE pour que pluginlib trouve le plugin
PLUGINLIB_EXPORT_CLASS(vehicle_plugins::Motorbike, vehicle_base::VehicleBase)